tests/testAllParallel.nim

```diff
@@ -0,0 +1,14 @@
+# import
+#   ./dht/[test_providers, test_providermngr],
+#   ./discv5/[test_discoveryv5, test_discoveryv5_encoding]
+
+import osproc
+
+let cmds = [
```

Comment:
it would be cool if we could just read all files from the top level `tests` directory that start with a `t` (I think this is how nimble does it as well) and just launch those, instead of having to manually adding them here?

Reply:
I was wondering a similar thing. Yah that's the default Nimble setup. 

Though there are two downsides I can think of: 
1. compilation takes longer than most nim projects, and I believe the Nimble approach compiles each separately? 
2. Splitting tests in half really helped speed up windows tests 

Both of those could probably be resolved. Thoughts though?

Reply:
Ah, actually I'd changed the setup here on a lost branch. For this sort of setup it definitely would work. Though I think the above issues are present. On Windows parallel tests didn't really speed it up -- I'm guessing due to extra compilation overhead on Windows?

Reply:
> On Windows parallel tests didn't really speed it up -- I'm guessing due to extra compilation overhead on Windows?

Hmm, interesting, inherently these tests are slow, because of the time needed for everything to settle (networking stack, connection establishment, etc...). Windows is notoriously slower than both linux and mac, both for compilation and the initial startup. But running things in parallel should help, so not sure :)

Reply:
I figured the Windows VMs are probably limited in vCPUs. Though it doesn't *slow* the tests down either.